test/validation/api/pool/pool.c

```diff
@@ -217,6 +217,51 @@ static void pool_test_info_packet(void)
 	CU_ASSERT(odp_pool_destroy(pool) == 0);
 }
 
+static void pool_test_info_data_range(void)
+{
+	odp_pool_t pool;
+	odp_pool_info_t info;
+	odp_pool_param_t param;
+	odp_packet_t pkt[PKT_NUM];
+	uint32_t i, num, seg_len;
+	uintptr_t pkt_data, pool_len;
+
+	odp_pool_param_init(&param);
+
+	param.type     = ODP_POOL_PACKET;
+	param.pkt.num  = PKT_NUM;
+	param.pkt.len  = PKT_LEN;
+
+	pool = odp_pool_create(NULL, &param);
+	CU_ASSERT_FATAL(pool != ODP_POOL_INVALID);
+
+	CU_ASSERT_FATAL(odp_pool_info(pool, &info) == 0);
+
+	pool_len = info.max_data_addr - info.min_data_addr + 1;
+	CU_ASSERT(pool_len >= PKT_NUM * PKT_LEN);
+
+	num = 0;
+
+	for (i = 0; i < PKT_NUM; i++) {
+		pkt[num] = odp_packet_alloc(pool, PKT_LEN);
+		CU_ASSERT(pkt[num] != ODP_PACKET_INVALID);
+
+		if (pkt[num] != ODP_PACKET_INVALID)
+			num++;
+	}
+
+	for (i = 0; i < num; i++) {
+		pkt_data = (uintptr_t)odp_packet_data(pkt[i]);
+		seg_len = odp_packet_seg_len(pkt[i]);
+		CU_ASSERT((pkt_data >= info.min_data_addr) &&
+			  (pkt_data + seg_len - 1 <= info.max_data_addr));
+
+		odp_packet_free(pkt[i]);
+	}
```

Comment:
Thanks to @psavol for catching the original problem, however this is still insufficient since every segment of the packet should be within bounds. A better test would look like:
```
for (i = 0; i < num; i++) {
        uint32_t pkt_len = odp_packet_len(pkt[i]);
        uint32_t offset = 0;

        while (offset < pkt_len) {
                pkt_data = (uintptr_t)odp_packet_offset(pkt[i], offset, &seg_len, NULL);
                CU_ASSERT((pkt_data >= info.min_data_addr) &&
                          (pkt_data + seg_len - 1 <= info.max_data_addr));
                offset += seg_len;
        }
        odp_packet_free(pkt[i]);
}
```